client/packages/system/src/InventoryAdjustmentReason/Components/InventoryAdjustmentReasonSearchInput.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react';
+import {
+  Autocomplete,
+  BasicTextInput,
+  Box,
+  defaultOptionMapper,
+  getDefaultOptionRenderer,
+  Typography,
+} from '@openmsupply-client/common';
+import {
+  useInventoryAdjustmentReason,
+  InventoryAdjustmentReasonRowFragment,
+} from '../api';
+
+interface InventoryAdjustmentReasonSearchInputProps {
+  value: InventoryAdjustmentReasonRowFragment | null;
+  width?: number | string;
+  onChange: (
+    inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null
+  ) => void;
+  autoFocus?: boolean;
+  stockReduction?: number;
+  isError?: boolean;
+}
+
+export const InventoryAdjustmentReasonSearchInput: FC<
+  InventoryAdjustmentReasonSearchInputProps
+> = ({
+  value,
+  width,
+  onChange,
+  autoFocus = false,
+  stockReduction,
+  isError,
+}) => {
+  const { data, isLoading } =
+    useInventoryAdjustmentReason.document.listAllActive();
+  const disabled = data?.totalCount === 0;
+
+  return (
```

Comment:
In my datafile I have some confusing names for the reasons, which led to errors when trying to save ( reducing stock and using a positive reason )

<img width="441" alt="Screenshot 2023-02-01 at 11 31 08 AM" src="https://user-images.githubusercontent.com/9192912/215905358-661dbe09-57bd-40fa-a57d-b28634254496.png">

![Screenshot 2023-02-01 at 11 38 11 AM](https://user-images.githubusercontent.com/9192912/215905350-5b773945-0e73-4496-9647-18e973cf8f25.png)

I've added `type` to the query, and used it to filter the reasons:

```
const reasonFilter = (reason: InventoryAdjustmentReasonRowFragment) => {
    if (stockReduction === 0 || !stockReduction) return true;
    if (stockReduction < 0)
      return reason.type === InventoryAdjustmentReasonNodeType.Positive;
    return reason.type === InventoryAdjustmentReasonNodeType.Negative;
  };
  const reasons = (data?.nodes ?? []).filter(reasonFilter);
  const options = defaultOptionMapper(reasons, 'reason');


/// then

        options={options}

```